### PR TITLE
[sailfish-office] Adapt to sharing UI and API changes. JB#53987 OMP#JOLLA-75

### PR DESCRIPTION
**Pull request by blammit**
The sharing UI is now launched out-of-process.

### CODE REVIEW

**Comment by blammit** [comment]
Note: associated API changes need to be merged before this can be merged.

File: qml/FileListPage.qml
Comment (chriadam) [inline qml/FileListPage.qml:302]: It previously used a serviceFilter "e-mail" as well as a mime type filter.  Was this change intentional?  Similarly for the other cases above.

Comment (blammit) [reply qml/FileListPage.qml:302]: Yes, it's okay to remove that now as that feature was only used to configure the in-app "add account" feature, but the user will now be sent to the Settings app so it's no longer necessary.

File: plugin/PDFDocumentPage.qml
Comment (pvuorela) [inline plugin/PDFDocumentPage.qml:201]: Seems to work, not fully sure why, but the sharing action by such only handling file paths, not file urls. But should be easy to add more explicit url support, i.e. if it starts with / or file://

**Comment by pvuorela** [comment]
LGTM assuming that the share api officially supports also urls.

File: plugin/PDFDocumentPage.qml
Comment (blammit) [reply plugin/PDFDocumentPage.qml]: Yes this was incorrect. Fixed now.

**Review by pvuorela — APPROVED** [review]

**Review by okodron — APPROVED** [review]

**Pull request merged**
